src/util/reader_mapping.cpp

```diff
@@ -203,6 +203,52 @@ ReaderMapping::get(const char* key, std::vector<unsigned int>& value,
 
 #undef GET_VALUES_MACRO
 
+bool
+ReaderMapping::get_compressed(const char* key, std::vector<unsigned int>& value,
+                              const std::optional<std::vector<unsigned int>>& default_value) const
+{
+  const auto sx = get_item(key);
+  if (!sx)
+  {
+    if (default_value)
+      value = *default_value;
+    return false;
+  }
+
+  assert_is_array(m_doc, *sx);
+  value.clear();
+  const auto& item = sx->as_array();
+  int multiplier = 0;
+  for (size_t i = 1; i < item.size(); ++i)
+  {
+    assert_is_integer(m_doc, item[i]);
+
+    const int val = item[i].as_int();
+    if (multiplier)
+    {
+      if (val < 0)
+      {
+        raise_exception(m_doc, item[i], "expected positive integer after multiplier");
+      }
+      value.insert(value.end(), multiplier, val);
+      multiplier = 0;
+    }
+    else if (val < 0)
+    {
+      multiplier = -val;
+    }
+    else
+    {
+      value.push_back(val);
```

Comment:
Could we possibly use `reserve` on whatever is going on here? Could minimalize some obsessive allocation calls.

Reply:
We can use `reserve` on the non-compressed functions for sure, but here, since I don't know the exact size at the end, should I just reserve the size of the uncompressed array and just deal with allocation calls whenever values need to be repeated?

Reply:
Sure, but from what im seeing, these repetitions happen quite often.

std vector and just preallocate it to like 4 once repetition happens. Its not a big deal actually, up to you.